spec/controllers/user_controller_spec.rb

```diff
@@ -1178,6 +1178,27 @@ def do_signin(email, password)
     expect(@user.profile_photo).not_to be_nil
   end
 
+  context 'there is no profile text' do
+    let(:user) { FactoryGirl.create(:user, :about_me => '') }
+
+    it 'prompts you to add profile text when adding a photo' do
+      session[:user_id] = user.id
+
+      profile_photo = ProfilePhoto.
+                        create(:data => load_file_fixture("parrot.png"),
+                               :user => user)
+
+      post :set_profile_photo, { :id => user.id,
+                                 :file => @uploadedfile,
+                                 :submitted_crop_profile_photo => 1,
+                                 :draft_profile_photo_id => profile_photo.id }
+
+      expect(flash[:notice]).
```

Comment:
elsewhere you tested the partial key explicitly, rather than the whole hash, it would be good to be consistent either way (I think just checking the partial is probably good where we can and there's no logic in that partial, since testing the text makes the spec more fragile).